docs/src/treasury.md

```diff
@@ -1 +1,120 @@
+
+1. Deploy the DeFiSpring contract
+2. Send tokens from the Treasury to the DeFiSpring contract
+3. Add a merkle root to the DeFiSpring contract
+
+This process is executed through an arbitrary proposal. Here's a step-by-step guide:
+
+### 1. Prepare the Arbitrary Proposal
+
+Create a new Cairo file `airdrop_controller.cairo` with the following structure:
+
+```rust
+#[starknet::interface]
+trait IAirdropController<TContractState> {
+    fn execute_arbitrary_proposal(ref self: TContractState);
+}
+
+#[starknet::contract]
+mod AirdropController {
+    use starknet::{ContractAddress, ClassHash};
+    use openzeppelin::token::erc20::interface::IERC20Dispatcher;
+    use konoha::treasury::ITreasuryDispatcher;
+    use defispring::IDefiSpringDispatcher;
+
+    #[storage]
+    struct Storage {}
+
+    #[external(v0)]
+    impl AirdropController of super::IAirdropController<ContractState> {
+        fn execute_arbitrary_proposal(ref self: ContractState) {
+            // Contract addresses (replace with actual addresses)
+            let governance_address: ContractAddress = 0x123...;
+            let treasury_address: ContractAddress = 0x456...;
+            let token_to_distribute: ContractAddress = 0x789...;
+            
+            // 1. Deploy DeFiSpring contract
+            let defispring_class_hash: ClassHash = 0xABC...;  // Replace with actual class hash
+            let mut calldata = ArrayTrait::new();
+            calldata.append(governance_address.into());  // Set governance as owner
+            let (defispring_address, _) = starknet::deploy_syscall(
+                defispring_class_hash, 0, calldata.span(), false
+            ).unwrap();
+
+            // 2. Send tokens from Treasury to DeFiSpring contract
+            let amount_to_distribute = 1000000000000000000000;  // Example: 1000 tokens
+            let treasury = ITreasuryDispatcher { contract_address: treasury_address };
+            treasury.send_tokens_to_address(defispring_address, amount_to_distribute, token_to_distribute);
+
+            // 3. Add merkle root to DeFiSpring contract
+            let merkle_root: felt252 = 0xDEF...;  // Replace with actual merkle root
+            let defispring = IDefiSpringDispatcher { contract_address: defispring_address };
+            defispring.add_root(merkle_root);
+        }
+    }
+}
+```
+
+### 2. Submit the Arbitrary Proposal
+
+Declare the AirdropController class and note its class hash.
+Submit a proposal with to_upgrade = 6 and the payload being the class hash of the AirdropController.
+
+### 3. Execute the Proposal
+
+Once the proposal passes, call apply_passed_proposal on the governance contract to execute the arbitrary proposal. This will:
```

Comment:
Could tag the [upgrades](docs/src/upgrades.md) documentation since it goes more in depth on apply_passed_proposal